components/core/cmake/Toolchains/utils.cmake

```diff
@@ -0,0 +1,40 @@
+function(setup_toolchains)
+    if("Darwin" STREQUAL "${CMAKE_HOST_SYSTEM_NAME}")
+        execute_process(
+            COMMAND
+                sw_vers --productVersion
+            OUTPUT_VARIABLE CMAKE_OSX_VERSION
```

Comment:
Let's not prefix the variable name with `CMAKE_` to avoid confusion. How about just `MACOS_VERSION`?